docs/proposals/106-spot-instance-karpenter/README.md

```diff
@@ -0,0 +1,450 @@
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this Proposal is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1: ML Engineer – Cost-Efficient Deployment of LLMs
+
+As a machine learning engineer deploying large language models (LLMs), I don't own any physical GPU servers, so I have to rent them from cloud providers. I want to automatically use cheaper GPU Spot instances for serving models when available, so that I can reduce infrastructure costs without sacrificing performance.
+
+#### Story 2: Workload Author – Flexible and Preferred GPU Scheduling with Spot Support
+
+As a workload author, I want to publish Kubernetes manifests for my model-serving workloads that are broadly compatible across different device types, without being overly prescriptive or requiring end users to modify them.
+
+- My workloads are GPU-dependent, but there are many different GPU models available in the cloud (e.g., A10, A100, H100).
+- Instead of locking my manifests to a single GPU type, I want to express a preference-ordered list of compatible GPU types (e.g., prefer A100, fall back to A10 or L4).
+- This gives end users the flexibility to run the same manifest on different underlying infrastructure.
+- If none of the existing nodes in the cluster meet the constraints (e.g., no compatible GPUs available), I want the system to automatically provision an appropriate Spot instance from the cloud provider, based on my declared GPU preferences and resource requirements.
+
+This approach allows me to build and share portable workloads that are cost-aware, device-flexible, and production-safe, without the need for users to rewrite manifests or manage instance-level complexity themselves.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+InftyAI ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
```

Comment:
Add one risk about the service availability once hitting the traffic spike, which will be tackled in beta graduation.